src/ip_who.rs

```diff
@@ -0,0 +1,37 @@
+use crate::{ConfigError, Location, LocationProvider};
+use reqwest::{
+  blocking::Client,
+  header::{HeaderMap, HeaderValue, CONTENT_TYPE},
+};
+
+// Fetch from ipwho.is
+pub struct IpWhoIs;
+
+impl LocationProvider for IpWhoIs {
+  fn location(&self) -> Result<Location, ConfigError> {
```

Comment:
This looks virtually identical to `IpInfoIoFetch`, so how about we extract a shared helper? e.g. `fn fetch(url) -> Location`

Reply:
we will have more fetcher in future which will have a different fetching mechanism